spec/unit/network/http/request_spec.rb

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env ruby
```

Comment:
Most of it was tested in `indirected_routes_spec`, but there are some routes, e.g. `puppet/v3/environments`, that create a `Request`, but are not routed to `indirected_routes`, since they don't handle indirected models. Since we're pushing more functionality into `Request` it seemed best to move the tests there too.